### PR TITLE
Drain errors in [E]GLCheck via loop

### PR DESCRIPTION
**Pull request by vittorioromeo**
Follow-up to #3233 and #3239. Didn't encounter any error in local testing, but let's see what CI has to say...

### CODE REVIEW

File: src/SFML/Graphics/GLCheck.hpp [hunk 1]
```diff
@@ -53,20 +58,28 @@ bool glCheckError(const std::filesystem::path& file, unsigned int line, std::str
...
-        }                                                                                        \
-        else                                                                                     \
-        {                                                                                        \
-            glCheckInternalFunction();                                                           \
-            sf::priv::glCheckError(__FILE__, static_cast<unsigned int>(__LINE__), #__VA_ARGS__); \
-        }                                                                                        \
+#define glCheck(...)                                                                                     \
+    [](auto&& glCheckInternalFunction)                                                                   \
+    {                                                                                                    \
+        if (const GLenum error = glGetError(); error != GL_NO_ERROR)                                     \
```
Comment (ZXShady) [inline src/SFML/Graphics/GLCheck.hpp:64]: choose a longer name like `glCheckInternalError` to not cause bugs with shadowing and consistency with the other names in the lamdba

File: src/SFML/Window/EGLCheck.hpp [hunk 2]
```diff
@@ -54,20 +59,28 @@ bool eglCheckError(const std::filesystem::path& file, unsigned int line, std::st
...
-        }                                                                                         \
-        else                                                                                      \
-        {                                                                                         \
-            eglCheckInternalFunction();                                                           \
-            sf::priv::eglCheckError(__FILE__, static_cast<unsigned int>(__LINE__), #__VA_ARGS__); \
-        }                                                                                         \
+#define eglCheck(...)                                                                                     \
+    [](auto&& eglCheckInternalFunction)                                                                   \
+    {                                                                                                     \
+        if (const EGLint error = eglGetError(); error != EGL_SUCCESS)                                     \
```
Comment (ZXShady) [inline src/SFML/Window/EGLCheck.hpp:65]: choose a longer name like eglCheckInternalError to not cause issues with shadowing.

**Review by ZXShady — CHANGES_REQUESTED** [review]

**Comment by ChrisThrasher** [comment]
SFMLBuildMaster: Build this please.

**Review by ChrisThrasher — COMMENTED** [review]
LGTM but I'm no OpenGL expert so I want to hear what @binary1248 has to say about it.

**Review by ChrisThrasher — APPROVED** [review]

**Review by ZXShady — APPROVED** [review]

**Pull request merged**